Add Autofill to data entry flow (#1528)
home-assistant/developers.home-assistant: docs/data_entry_flow_index.md

        """Handle user step."""
```

> Data entry flows depend on translations for showing the text in the steps. It depends on the parent of a data entry flow manager where this is stored. For config and option flows this is in `strings.json` under `config` and `option`, respectively.

Reviewer: ```suggestion
Data entry flows depend on translations for showing the text in the steps. It depends on the parent of a data entry flow manager where this is stored. For config and option flows, this is in `strings.json` under `config` and `option`, respectively.
```

> Data entry flows depend on translations for showing the text in the steps. It depends on the parent of a data entry flow manager where this is stored. For config and option flows this is in `strings.json` under `config` and `option`, respectively  .

Reviewer: ```suggestion
Data entry flows depend on translations for showing the text in the steps. It depends on the parent of a data entry flow manager where this is stored. For config and option flows this is in `strings.json` under `config` and `option`, respectively.
```

|   `description`    | Form instructions  | Optional. Do not link to the documentation as that is linked automatically. Do not include "basic" information like "Here you can set up X". |
|       `data`       |    Field labels    | Keep succinct and consistent with other integrations whenever appropriate for the best user experience.                                      |
| `data_description` | Field descriptions | Optional explanatory text to show below the field.                                                                                           |

The field labels and descriptions are given as a dictionary with keys corresponding to your schema. Here is a simple example:

```json
{
  "config": {
    "step": {
      "user": {
          "title": "Add Group",
          "description": "Some description",
          "data": {
              "entities": "Entities",
          },
          "data_description": {
              "entities": "The entities to add to the group",
          },
      }
    }
  }
}
```

#### Enabling Browser Autofill

> If your integration is collecting form data which can be automatically filled by browsers or password managers, such as login credentials or contact information, then you have 2 options to enable this feature.

Reviewer: ```suggestion
Suppose your integration collects form data that can be automatically filled by browsers or password managers, such as login credentials or contact information. In that case, you have two options to enable this feature.
```

Author: I agree the sentence was too long, but it should sound more imperative than this or even the way I wrote it.  Try the change I pushed.

> The first option is to use Voluptuous with data keys that are recognized by the frontend. This is limited to username and password fields, and is supported primarily to quickly enable auto-fill on the many integrations that collect them without having to convert their schemas to selectors. The frontend will recognize the keys `"username"` and `"password"` and add HTML `autocomplete` attribute values of `"username"` and `"current-password"`, respectively.

Reviewer: ```suggestion
The first option is to use Voluptuous with data keys recognized by the frontend. The frontend will recognize the keys `"username"` and `"password"` and add HTML `autocomplete` attribute values of `"username"` and `"current-password"` respectively. Support for autocomplete is limited to `"username"` and `"password"` fields and is supported primarily to quickly enable auto-fill on the many integrations that collect them without converting their schemas to selectors.
```

> The second option is to use a [text selector](https://www.home-assistant.io/docs/blueprint/selectors/#text-selector). This gives full control of the input type and allows any permitted value for `autocomplete` to be specified. A hypothetical schema collecting certain fillable data might be:

Reviewer: ```suggestion
The second option is to use a [text selector](https://www.home-assistant.io/docs/blueprint/selectors/#text-selector). A text selector gives full control of the input type and allows any permitted value for `autocomplete` to be specified. A hypothetical schema collecting specific fillable data might be:
```

        )
```

#### Multi-step flows

> If the user input passes validation, you can again return one of the possible step types. If you want to navigate the user to the next step, return the return value of that step:

Reviewer: ```suggestion
If the user input passes validation, you can return one of the possible step types again. If you want to navigate the user to the next step, return the return value of that step:
```
